h/api/resources.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.security import Allow, Authenticated, ALL_PERMISSIONS
+
+from .models import Annotation
+
+
+class Resource(dict):
+    """
+    Resource is an entry in the traversal tree.
+    """
+    __name__ = None
+    __parent__ = None
+
+    def __init__(self, request):
+        pass
+
+    def add(self, name, obj):
+        obj.__name__ = name
+        obj.__parent__ = self
+        self[name] = obj
+
+
+class Annotations(object):
+    """
+    Annotations is a container resource that exposes annotations as its
+    children in the tree.
+    """
+    def __getitem__(self, key):
+        instance = Annotation.fetch(key)
+        if instance is None:
+            raise KeyError(key)
+        instance.__name__ = key
+        instance.__parent__ = self
+        return instance
+
+
+class Root(Resource):
+    __acl__ = [
+        (Allow, 'group:admin', ALL_PERMISSIONS),
+        (Allow, Authenticated, 'create'),
+    ]
+
+    def __init__(self, request):
+        self.add('annotations', Annotations())
```

Comment:
For consistency, shouldn't `Annotations` be a `Resource` and take a `request` to its constructor?  Or, better yet, remove the constructor from the base `Resource` and let's maybe make an explicit root factory function.

```python
def root_factory(request):
    return Root(annotations=Annotations())
```